src/main/java/com/solace/samples/java/patterns/GuaranteedProcessor.java

```diff
@@ -0,0 +1,227 @@
+            msgRecvCounter = 0;
+            msgSentCounter = 0;
+        }, 1, 1, TimeUnit.SECONDS);
+
+        
+        while (System.in.available() == 0 && !isShutdown) {
+            InboundMessage inboundMsg = receiver.receiveMessage(1000);  // blocking receive a message
+            if (inboundMsg == null) {  // receive() either got interrupted, or timed out
+                continue;
+            }
+            msgRecvCounter++;
+            String inboundTopic = inboundMsg.getDestinationName();
+            if (inboundTopic.contains("/pers/pub/")) {  // simple validation of topic
+                // how to "process" the incoming message? maybe do a DB lookup? add some additional properties? or change the payload?
+                OutboundMessageBuilder messageBuilder = messagingService.messageBuilder();
+                messageBuilder.withApplicationMessageId(inboundMsg.getApplicationMessageId());  // set the new message ID to the same as this one
+                final String upperCaseTopic = inboundTopic.toUpperCase();  // as a silly example of "processing"
+                OutboundMessage outboundMsg = messageBuilder.build(upperCaseTopic);
+                String [] inboundTopicLevels = inboundTopic.split("/",6);
+                String onwardsTopic = new StringBuilder(TOPIC_PREFIX).append(API.toLowerCase())
+                		.append("/pers/upper/").append(inboundTopicLevels[5]).toString();
+                try {
+                	ProcessorCorrelationKey ck = new ProcessorCorrelationKey(inboundMsg, outboundMsg, receiver);
+                	publisher.publish(outboundMsg, Topic.of(onwardsTopic), ck);
+                    msgSentCounter++;
+                } catch (RuntimeException e) {  // threw from publish(), only thing that is throwing here, but keep trying (unless shutdown?)
+                    logger.warn("### Caught while trying to publisher.publish()",e);
+                    isShutdown = true;  // just example, maybe look to see if recoverable
+                }
+            } else {  // unexpected message. either log or something
+                logger.info("Received an unexpected message with topic "+inboundTopic+".  Ignoring");
+                receiver.ack(inboundMsg);
+            }
+        }
+
+
+        // async queue receive working now, so time to wait until done...
+        System.out.println(SAMPLE_NAME + " connected, and running. Press [ENTER] to quit.");
+        while (System.in.available() == 0 && !isShutdown) {
+            Thread.sleep(1000);  // wait 1 second
+            System.out.printf("%s %s Received msgs/s: %,d%n",API,SAMPLE_NAME,msgRecvCounter);  // simple way of calculating message rates
+            msgRecvCounter = 0;
+            if (hasDetectedRedelivery) {  // try shutting -> enabling the queue on the broker to see this
+                System.out.println("*** Redelivery detected ***");
+                hasDetectedRedelivery = false;  // only show the error once per second
```

Comment:
it doesn't look like this could ever have been set to true in this sample. remove to keep it simple?